File: src/desktop/ui/src/Analysis/Models/ReadingCitation.ts
import type { ReadingFocusRange } from "./ReadingFocusRange";
import type { ReadingSide } from "./ReadingSide";
import type { ReadingTrust } from "./ReadingTrust";

export interface ReadingCitation {
  readonly claimId: string;
  readonly nodeId: string;
  readonly side: ReadingSide;
  readonly path: string;
  readonly objectId: string;
  readonly startLine: number;
  readonly endLine: number;
  readonly focus: readonly ReadingFocusRange[];
  readonly provenance: ReadingTrust;
}

File: src/desktop/src-tauri/src/analysis/models/reading_citation.rs
use crate::analysis::models::{ReadingFocusRange, ReadingSide, ReadingTrust};
use serde::{Deserialize, Serialize};

/// Represents a citation that anchors a claim to a source location.
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ReadingCitation {
    pub claim_id: String,
    pub node_id: String,
    pub side: ReadingSide,
    pub path: String,
    pub object_id: String,
    pub start_line: u32,
    pub end_line: u32,
    pub focus: Vec<ReadingFocusRange>,
    pub provenance: ReadingTrust,
}
